### PR TITLE
Feature 1

### PR DESCRIPTION
**Pull request by mirontoli**
This is code for feature 1: getting small images in lists when available. It uses [ImageDownloader](http://code.google.com/p/android-imagedownloader/source/browse/trunk/src/com/example/android/imagedownloader/ImageDownloader.java) which is described on the [android blog](http://android-developers.blogspot.se/2010/07/multithreading-for-performance.html). The image downloader uses cache for performance.

screencap
# Some improvements can be made:
- a default image for images which are beeing downloaded or, if they not exist (for now it is a question mark image).
- ...

### CODE REVIEW

File: Java/Malmofestivalen/src/dk/christer/malmofestivalen/EventsForCategoryActivity.java [hunk 1]
```diff
@@ -71,29 +62,15 @@ private void createEventsList() {
...
 				this, 
 		        R.layout.eventitemrow,  
 		        mCursor,                
 		        new String[] {EventProvider.EVENT_KEY_TITLE, EventProvider.EVENT_KEY_SCENETITLE, EventProvider.EVENT_KEY_STARTDATE},
-		        new int[] {R.id.eventitemrowtitle, R.id.eventitemrowscene, R.id.eventitemrowtimeresume});  
-		
-        	adapter.setViewBinder(new SimpleCursorAdapter.ViewBinder() {
-				
-				@Override
-				public boolean setViewValue(View view, Cursor event, int columnIndex) {
```
Comment (mirontoli) [inline Java/Malmofestivalen/src/dk/christer/malmofestivalen/EventsForCategoryActivity.java:84]: Now the values are rendered in custom adapter (EventCursorAdapter)

**Comment by christerdk** [comment]
I've pulled down the code and tried to compile. This line gives me a compile error (in EventCursorAdapter):

ImageView iv = (ImageView) row.findViewById(R.id.eventitemrowimage);

Could it be that changes to eventitemrow.xml (or some other layout) did not get included in this pull request?

**Comment by mirontoli** [comment]
Yes. I missed to commit eventitemrow.xml. I'll do it soon.

**Comment by christerdk** [comment]
Sure, no problem. Enjoy the Sunday, we're not that busy :)

On 22/04/2012, at 09.28, Anatoly Mironov
reply@reply.github.com
wrote:

**Comment by christerdk** [comment]
Really nice job!

A couple of suggestions that I think I will go forward with is switching to async download (the performance is crippled a a bit by sync download), increasing the amount of objects in the cache (now 10), and also the purge time (now only 10 sec). I'll switch these things in my feature_1 branch and then merge to master, and then we'll take any follow bugs from there.

Once again, nice job!

**Comment by mirontoli** [comment]
Thank you. It was fun to write some code for android again. It was a while since the last time. I agree there are some improvement that can be made, and we will do them. The imageDownloader was a nice class which surely can be used on other activities.

**Comment by christerdk** [comment]
Absolutely. It could simplify the details view..

**Comment by christerdk** [comment]
We have an observation with regards to download of pictures. Is this something you could look into?

https://github.com/christerdk/Malmofestivalen-for-Android/issues/7

**Comment by mirontoli** [comment]
I'll look att it in the weekend.

**Pull request merged**